arch/Kconfig

```diff
@@ -339,6 +339,9 @@ config ARCH_HAS_EXECUTABLE_PAGE_BIT
 config ARCH_HAS_NOCACHE_MEMORY_SUPPORT
 	bool
 
+config ARCH_HAS_RAMFUNC_SUPPORT
```

Comment:
@andrewboie are you ok with this cross-arch Kconfig option?

Reply:
No not really, my issue is with the name.
Because this Kconfig is describing an architecture capability, and if the architecture is capable, then we automatically use up a static memory region for the ramfunc area even if there is nothing in it, this gets selected by another Kconfig.

Just call this CONFIG_RAMFUNC, and have it depend on CONFIG_XIP, and let the user turn it on if they want to.

Now it IS true that some XIP boards (like arduino 101) are incapable from fetching instructions from anywhere but flash for security reasons, but I don't think we really need a kconfig ontology for that.

Reply:
> Because this Kconfig is describing an architecture capability, and if the architecture is capable, then we automatically use up a static memory region for the ramfunc area even if there is nothing in it, this gets selected by another Kconfig.

That is not correct. If __ramfunc is not used, _ramfunc_ram_size is 0. In that case the MPU region is not configured (that's a nice side effect of the MPU rewrite), so no region is wasted.

Reply:
OK that works for me